osquery/tables/system/darwin/signature.mm

```diff
@@ -58,130 +62,159 @@ Status getVerifyFlags(SecCSFlags& flags) {
-            ss << (unsigned int)bytes[n];
-          }
-          r["cdhash"] = ss.str();
-        }
-        if (r["cdhash"].length() != CC_SHA1_DIGEST_LENGTH * 2) {
-          VLOG(1) << "Error extracting code directory hash";
-          r["cdhash"] = "";
-        }
-      }
+  // If we don't get an identifier for this file, then it's not signed.
+  CFStringRef ident =
+      (CFStringRef)CFDictionaryGetValue(code_info, kSecCodeInfoIdentifier);
 
-      // Team Identifier
-      r["team_identifier"] = "";
-      CFTypeRef teamIdent = nullptr;
-      if (CFDictionaryGetValueIfPresent(codeInfo, kSecCodeInfoTeamIdentifier,
-                                        &teamIdent)) {
-        r["team_identifier"] = stringFromCFString((CFStringRef)teamIdent);
+  if (ident == nullptr) {
+    CFRelease(code_info);
+    results.push_back(r);
+    CFRelease(static_code);
+    return Status(1, "No identifier found for file(" + arch + "): " + path);
+  }
+
+  r["identifier"] = stringFromCFString(ident);
+
+  // Get CDHash
+  r["cdhash"] = "";
+  CFDataRef hashInfo =
+      (CFDataRef)CFDictionaryGetValue(code_info, kSecCodeInfoUnique);
+  if (hashInfo != nullptr) {
+    r["cdhash"].reserve(CC_SHA1_DIGEST_LENGTH);
+    // Get the SHA-1 bytes
+    std::stringstream ss;
+    auto bytes = CFDataGetBytePtr(hashInfo);
+    if (bytes != nullptr &&
+        CFDataGetLength(hashInfo) == CC_SHA1_DIGEST_LENGTH) {
+      // Write bytes as hex strings
+      for (size_t n = 0; n < CC_SHA1_DIGEST_LENGTH; n++) {
+        ss << std::hex << std::setfill('0') << std::setw(2);
+        ss << (unsigned int)bytes[n];
       }
+      r["cdhash"] = ss.str();
+    }
+    if (r["cdhash"].length() != CC_SHA1_DIGEST_LENGTH * 2) {
+      VLOG(1) << "Error extracting code directory hash";
+      r["cdhash"] = "";
+    }
+  }
 
-      // Get common name
-      r["authority"] = "";
-      CFArrayRef certChain =
-          (CFArrayRef)CFDictionaryGetValue(codeInfo, kSecCodeInfoCertificates);
-      if (certChain != nullptr && CFArrayGetCount(certChain) > 0) {
-        auto cert = SecCertificateRef(CFArrayGetValueAtIndex(certChain, 0));
-        auto der_encoded_data = SecCertificateCopyData(cert);
-        if (der_encoded_data != nullptr) {
-          auto der_bytes = CFDataGetBytePtr(der_encoded_data);
-          auto length = CFDataGetLength(der_encoded_data);
-          auto x509_cert = d2i_X509(nullptr, &der_bytes, length);
-          if (x509_cert != nullptr) {
-            std::string subject;
-            std::string issuer;
-            std::string commonName;
-            genCommonName(x509_cert, subject, commonName, issuer);
-            r["authority"] = commonName;
-            X509_free(x509_cert);
-          } else {
-            VLOG(1) << "Error decoding DER encoded certificate";
-          }
-          CFRelease(der_encoded_data);
-        }
+  // Team Identifier
+  r["team_identifier"] = "";
+  CFTypeRef teamIdent = nullptr;
+  if (CFDictionaryGetValueIfPresent(
```

Comment:
I feel like this is the best way to use these APIs.
- Check if the value is present and get the `CFTypeRef`.
- Check if the type is the type you expect (using CF APIs for comparison)
- Explicit cast and use that value inside the scope.